presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergLogicalPlanner.java

```diff
@@ -77,6 +102,137 @@ protected QueryRunner createQueryRunner()
         return createIcebergQueryRunner(ImmutableMap.of("experimental.pushdown-subfields-enabled", "true"), ImmutableMap.of());
     }
 
+    @Test
+    public void testFilterPushdown()
+    {
+        Session pushdownFilterEnabled = pushdownFilterEnabled();
+
+        // Only domain predicates
+        assertPlan("SELECT linenumber FROM lineitem WHERE partkey = 10",
+                output(exchange(project(
+                        filter("partkey=10",
+                                strictTableScan("lineitem", identityMap("linenumber", "partkey")))))));
+
+        assertPlan(pushdownFilterEnabled, "SELECT linenumber FROM lineitem WHERE partkey = 10",
+                output(exchange(
+                        strictTableScan("lineitem", identityMap("linenumber")))),
+                plan -> assertTableLayout(
+                        plan,
+                        "lineitem",
+                        withColumnDomains(ImmutableMap.of(new Subfield(
+                                "partkey",
+                                ImmutableList.of()),
+                                singleValue(BIGINT, 10L))),
+                        TRUE_CONSTANT,
+                        ImmutableSet.of("partkey")));
+
+        assertPlan(pushdownFilterEnabled, "SELECT partkey, linenumber FROM lineitem WHERE partkey = 10",
+                output(exchange(
+                        strictTableScan("lineitem", identityMap("partkey", "linenumber")))),
+                plan -> assertTableLayout(
+                        plan,
+                        "lineitem",
+                        withColumnDomains(ImmutableMap.of(new Subfield(
+                                "partkey",
+                                ImmutableList.of()),
+                                singleValue(BIGINT, 10L))),
+                        TRUE_CONSTANT,
+                        ImmutableSet.of("partkey")));
+
+        // Only remaining predicate
+        assertPlan("SELECT linenumber FROM lineitem WHERE mod(orderkey, 2) = 1",
```

Comment:
I think it's better to move this test to be together the counterpart with filter pushdown enabled later.